gp-includes/things/project.php

```diff
@@ -233,10 +233,25 @@ public function regenerate_paths( $parent_project_id = null ) {
 	}
 
 	public function source_url( $file, $line ) {
-		if ( $this->source_url_template() ) {
-			return str_replace( array('%file%', '%line%'), array($file, $line), $this->source_url_template() );
+		$source_url = false;
+
+		if ( $source_url_template = $this->source_url_template() ) {
+			$source_url = str_replace( array( '%file%', '%line%' ), array( $file, $line ), $source_url_template );
 		}
-		return false;
+
+		/**
+		 * Allows per-reference overriding of the source URL defined as project setting.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param string        $source_url   The originally generated source URL.
+		 * @param GP_Project    $project      The current project.
+		 * @param string        $file         The referenced file name.
+		 * @param string        $line         The line number in the referenced file.
```

Comment:
There should be only one space between `GP_Project` and `$project`. Same for `$source_url` and its descriptions. The other parameters should be aligned accordingly.

`$source_url` can be false too:
`@param string|false $source_url The originally generated source URL, or false if no URL is available.`